core/src/main/java/io/methvin/watcher/DirectoryWatcher.java

```diff
@@ -200,6 +247,16 @@ public void watch() {
           }
           Path childPath = eventPath == null ? null : keyRoots.get(key).resolve(eventPath);
           logger.debug("{} [{}]", kind, childPath);
+
+          // If a change in a directory of a non-recursive directory happens, ignore it
+          if (Files.isDirectory(childPath, NOFOLLOW_LINKS)) {
+            Path childParentPath = childPath.getParent();
```

Comment:
Is it safe to assume here that if I do mkdirs `foo/bar/baz` and `foo` is a non-recursive directory, I'll get the event about the creation of `foo/bar` before the event about `foo/bar/baz`? Otherwise this implementation is flawed.